app/src/main/java/org/schabi/newpipe/player/PlayerService.kt

```diff
@@ -148,203 +150,204 @@ public int onStartCommand(final Intent intent, final int flags, final int startI
         }
 
-        if (BIND_PLAYER_HOLDER_ACTION.equals(intent.getAction())) {
+        if (BIND_PLAYER_HOLDER_ACTION == intent.action) {
             // Note that this binder might be reused multiple times while the service is alive, even
             // after unbind() has been called: https://stackoverflow.com/a/8794930 .
-            return mBinder;
-
-        } else if (MediaBrowserServiceCompat.SERVICE_INTERFACE.equals(intent.getAction())) {
+            return mBinder
+        } else if (SERVICE_INTERFACE == intent.action) {
             // MediaBrowserService also uses its own binder, so for actions related to the media
             // browser service, pass the onBind to the superclass.
-            return super.onBind(intent);
-
+            return super.onBind(intent)
         } else {
             // This is an unknown request, avoid returning any binder to not leak objects.
-            return null;
+            return null
         }
     }
 
-    public static class LocalBinder extends Binder {
-        private final WeakReference<PlayerService> playerService;
+    class LocalBinder internal constructor(playerService: PlayerService) : Binder() {
+        private val playerService: WeakReference<PlayerService?>
 
-        LocalBinder(final PlayerService playerService) {
-            this.playerService = new WeakReference<>(playerService);
+        init {
+            this.playerService = WeakReference<PlayerService?>(playerService)
         }
 
-        public PlayerService getService() {
-            return playerService.get();
-        }
-    }
-
-    /**
-     * @return the current active player instance. May be null, since the player service can outlive
-     * the player e.g. to respond to Android Auto media browser queries.
-     */
-    @Nullable
-    public Player getPlayer() {
-        return player;
+        val service: PlayerService?
+            get() = playerService.get()
     }
 
     /**
      * Sets the listener that will be called when the player is started or stopped. If a
-     * {@code null} listener is passed, then the current listener will be unset. The parameter taken
-     * by the {@link Consumer} can be null to indicate that the player is stopping.
+     * `null` listener is passed, then the current listener will be unset. The parameter taken
+     * by the [Consumer] can be null to indicate that the player is stopping.
      * @param listener the listener to set or unset
      */
-    public void setPlayerListener(@Nullable final Consumer<Player> listener) {
-        this.onPlayerStartedOrStopped = listener;
+    fun setPlayerListener(listener: Consumer<Player?>?) {
+        this.onPlayerStartedOrStopped = listener
         if (listener != null) {
             // if there is no player, then `null` will be sent here, to ensure the state is synced
-            listener.accept(player);
+            listener.accept(player)
         }
     }
-    //endregion
 
+    //endregion
     //region Media browser
```

Comment:
Why are all of these swapped?

Reply:
Are they? It’s the previous region that ends and the new one that starts

Reply:
Yeah and there's space in the middle, but anyway, this isn't important